service/history/workflow/update/registry.go

```diff
@@ -195,10 +196,18 @@ func (r *registry) Find(ctx context.Context, id string) (*Update, bool) {
 	return r.findLocked(ctx, id)
 }
 
-func (r *registry) TerminateUpdates(_ context.Context, _ EventStore) {
-	// TODO (alex-update): implement
-	// This method is not implemented and update API callers will just timeout.
-	// In future, it should remove all existing updates and notify callers with better error.
+// CancelIncomplete cancels all incomplete updates in the registry:
+//   - updates in stateAdmitted, stateRequested, or stateSent are rejected,
+//   - updates in stateAccepted are ignored (see CancelIncomplete() in update.go for details),
+//   - updates in stateCompleted are ignored.
+func (r *registry) CancelIncomplete(ctx context.Context, reason CancelReason, eventStore EventStore) error {
```

Comment:
I wonder if there's some central place that allow us to perform this operation on workflow close, so we don't have to remember to call it on every path that lead to workflow close.

Reply:
Moreover, there is no single place even for all cases where we terminate WF. This why I postponed these changes to another PR. And there is also WF timeout case. I will create a task to refactor this.

Reply:
hmm if mutable state has access to the registry then probably we can perform the operation there. 🤔